src/OpenApi/src/Schemas/OpenApiJsonSchema.Helpers.cs

```diff
@@ -0,0 +1,284 @@
+                reader.Read();
+                obj[key] = ReadOpenApiAny(ref reader);
+                reader.Read();
+            }
+            return obj;
+        }
+
+        return default;
+    }
+
+    /// <summary>
+    /// Read a property node from the given JSON reader instance.
+    /// </summary>
+    /// <param name="reader">The <see cref="Utf8JsonReader"/> to consume the property value from.</param>
+    /// <param name="propertyName">The name of the property the editor is currently consuming.</param>
+    /// <param name="schema">The <see cref="OpenApiSchema"/> to write the given values to.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> instance.</param>
+    public static void ReadProperty(ref Utf8JsonReader reader, string propertyName, OpenApiSchema schema, JsonSerializerOptions options)
+    {
+        switch (propertyName)
+        {
+            case "type":
+                reader.Read();
+                if (reader.TokenType == JsonTokenType.StartArray)
+                {
+                    var types = ReadList<string>(ref reader, options);
+                    foreach (var type in types ?? [])
+                    {
+                        // JSON Schema represents nullable types using an array consisting of
+                        // the target type and "null". Since OpenAPI Schema does not support
+                        // representing types within an array we must check for the "null" type
+                        // and map it to OpenAPI's `nullable` property for OpenAPI v3.
+                        if (type == "null")
+                        {
+                            schema.Nullable = true;
+                        }
+                        else
+                        {
+                            schema.Type = type;
+                        }
+                    }
+                }
+                else
+                {
+                    var type = reader.GetString();
+                    schema.Type = type;
+                }
+                break;
+            case "enum":
+                reader.Read();
+                var enumValues = ReadList<string>(ref reader, options);
+                if (enumValues is not null)
+                {
+                    schema.Enum = enumValues.Select(v => new OpenApiString(v)).ToList<IOpenApiAny>();
+                }
+                break;
+            case "default":
+                reader.Read();
+                schema.Default = ReadOpenApiAny(ref reader);
+                break;
+            case "items":
+                reader.Read();
+                var valueConverter = (JsonConverter<OpenApiJsonSchema>)options.GetTypeInfo(typeof(OpenApiJsonSchema)).Converter;
+                schema.Items = valueConverter.Read(ref reader, typeof(OpenApiJsonSchema), options)?.Schema;
+                break;
+            case "nullable":
+                reader.Read();
+                schema.Nullable = reader.GetBoolean();
```

Comment:
Last one wins, compared to `type: null` above?

Reply:
`type: ["string", null]` and `type: "string", nullable: true` are mutually exclusive. `nullable` is only set by us in the `ApplyPrimitiveFormat` call that is applied via the `OnGenerated` callback where we'll modify the type so it doesn't fall into the null case above.